lib/linguist/heuristics.yml

```diff
@@ -453,16 +453,15 @@ disambiguations:
     pattern: '(?i:^\\i\b|AS \$\$|LANGUAGE ''?plpgsql''?|BEGIN( WORK )?;)'
   # IBM db2
   - language: SQLPL
-    pattern: "(?i:alter module|MODE DB2SQL)"
+    pattern: '(?i:alter module|MODE DB2SQL|\bSYSCAT\.\b|\bSYSPROC\.\b|ASSOCIATE RESULT SET|\bEND!\s*$)'
   # Oracle
   - language: PLSQL
     pattern: '(?i:\$\$PLSQL_|XMLTYPE|systimestamp|\.nextval|connect by|AUTHID (DEFINER|CURRENT_USER)|constructor\W+function)'
   # T-SQL
   - language: TSQL
-    pattern: '(?i:^\s*GO\b|BEGIN( TRY| CATCH)|OUTPUT INSERTED|DECLARE\s+@)'
-  # Generic SQL
+    pattern: '(?i:^\s*GO\b|BEGIN( TRY| CATCH)|OUTPUT INSERTED|DECLARE\s+@|\[dbo\])'
+  # Fallback to SQL
```

Comment:
Not sure we even need this comment; it's fairly obvious it's an open-ended fallback.

Reply:
Removed.